polpo/lddmm/parallel_transport.py

```diff
@@ -0,0 +1,213 @@
+    deformation="geodesic",
+    external_forces=None,
+    use_rk2_for_flow=False,
+    use_rk2_for_shoot=False,
+):
+    """Exponential map.
+
+    Compute the deformation of a source shape by the flow parametrized by control points and
+    momenta.
+
+    Parameters
+    ----------
+    source: str or pathlib.Path
+        Path to the vtk file that contains the source mesh.
+    control_points: str or pathlib.Path
+        Path to the txt file that contains the initial control points.
+    momenta: str or pathlib.Path
+        Path to the txt file that contains the initial momenta.
+    kernel_width: float
+        Width of the Gaussian kernel. Controls the spatial smoothness of the deformation and
+        influences the number of parameters required to represent the deformation.
+        Optional, default: 20.
+    kernel_type: str, {torch, keops}
+        Package to use for convolutions of velocity fields and loss functions.
+    kernel_device: str, {cuda, cpu}
+    regularisation: unused.
+    write_params: bool
+    external_forces: str or pathlib.Path
+        Path to the vtk file that contains the external forces to compute a spline deformation.
+    use_rk2_for_flow: bool
+        Whether to use Runge-Kutta order 2 steps in the integration of the flow equation, i.e. when
+        warping the shape. If False, a Euler step is used.
+        Optional, default: False
+    use_rk2_for_shoot: bool
+        Whether to use Runge-Kutta order 2 steps in the integration of the Hamiltonian equation that
+        governs the time evolution of control points and momenta. If False, a Euler step is used.
+        Optional, default: False
+    """
+    deformation_parameters = {
+        "deformation_model": deformation,
+        "deformation_kernel_type": kernel_type,
+        "deformation_kernel_width": kernel_width,
+        "deformation_kernel_device": kernel_device,
+        "concentration_of_time_points": number_of_time_steps,
+        "number_of_time_points": number_of_time_steps + 1,
+        "use_rk2_for_flow": use_rk2_for_flow,
+        "use_rk2_for_shoot": use_rk2_for_shoot,
+        "output_dir": output_dir,
+        "write_adjoint_parameters": write_params,
+    }
+
+    template_specifications = {
+        "shape": {
+            "deformable_object_type": "landmark",
+            "kernel_type": kernel_type,
+            "kernel_width": kernel_width,
+            "kernel_device": kernel_device,
+            "noise_std": regularisation,
+            "filename": source,
+            "noise_variance_prior_scale_std": None,
+            "noise_variance_prior_normalized_dof": 0.01,
+        }
+    }
+
+    Deformetrica(output_dir, verbosity="INFO")
+    compute_shooting(
+        template_specifications,
+        initial_control_points=control_points,
+        external_forces=external_forces,
+        initial_momenta=momenta,
+        **deformation_parameters,
+    )
+
+    return time.gmtime()
+
+def estimate_parallel_transport(
```

Comment:
what's the difference between this and `transport`?

Reply:
nice. In geomstats both functions are under `parallel_transport` (with different arguments). but I agree it is nicer to split it.

I'd find better names though. `estimate_parallel_transport` seems overkill when we have a method called `shoot`. i.e. I'd probably use verbs like `estimate` in the learning part, but not in `geometry`, a bit following `geomstats`